### PR TITLE
feat: Swapped out centos for alpine

### PR DESCRIPTION
**Pull request by cagiti**
Makes the jx image a little lighter :)

### CODE REVIEW

**Comment by tdcox** [comment]
Is this going to have a knock-on effect to the other builder images?

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,9 @@
-FROM centos:7
+FROM alpine:3.10
```
Comment (hferentschik) [inline Dockerfile]: +1, I love the idea of switching to Alpine. I am using Alpine myself when testing images and have a very similar Dockerfile. It makes building and pushing custom images more lightweight. For what it's worth I am using:

```
FROM alpine:3.10

RUN apk --update add ca-certificates git

COPY ./build/linux/jx /usr/bin/jx
```

I find the current `ENTRYPOINT` just gets into the way when using the image. I'd remove it.

One thing to keep in mind here is that we might want to try to align the used based images. @jstrachan and I had a discussion around this the other day and James was interested in somehow trying to align on a common base image. Also, he mentioned that centos was originally used due to some issues (DNS!?) with Alpine. For my part I have not seen any problems yet.

Comment (tdcox) [reply Dockerfile]: Alpine has had horrendous issues relating to DNS resolution within Kubernetes that they never seemed to be able to resolve and generally blamed upon “dependencies outside their control”.

Alpine is also not compatible with certain physical infrastructure in the cloud, such as GPU cards, due to a lack of driver support. For this reason, the machine learning builders require Centos.

Comment (hferentschik) [reply Dockerfile]: > Alpine has had horrendous issues relating to DNS resolution within Kubernetes that they never seemed to be able to resolve and generally blamed upon “dependencies outside their control”.

I guess you are referring to https://github.com/gliderlabs/docker-alpine/issues/255. I was not aware of this issue and have not noticed it before. 

I wonder though if it is really alpine specific. I wonder whether the issues we seeing at times to clone a git repo are not also related to DNS issues. But that's just a guess.

That said, I am not bound to alpine at all. I just would like to have a more leight weight default image.

> Alpine is also not compatible with certain physical infrastructure in the cloud, such as GPU cards, due to a lack of driver support. For this reason, the machine learning builders require Centos.

Sure, there are always special needs. But in the end you can always use custom images, right? 

But maybe this should be discussed on an issue instead.

Comment (hferentschik) [reply Dockerfile]: I opened https://github.com/jenkins-x/jx/issues/4481

Comment (cagiti) [reply Dockerfile]: > Is this going to have a knock-on effect to the other builder images?

It shouldn't, I don't believe this is used as a base image

Comment (hferentschik) [reply Dockerfile]: How do we move forward with this PR?

**Comment by hferentschik** [comment]
@cagiti shall we close this PR for now?

**Pull request closed without merging**